### PR TITLE
Pbi 93 seed vendor table

### PR DESCRIPTION
**Pull request by valenciusap17**
## Describe your changes

- Add fixtures for seeding data
- Implement seederWriter for **Vendor** and **ProductVendor**
- Add relevant Unit Test for seederWriter of **Vendor** and **ProductVendor**

## Issue ticket number and link
ADU-93: https://linear.app/adudu/issue/ADU-93/seeder-for-vendor-and-product-vendor
ADU-131: https://linear.app/adudu/issue/ADU-131/[unit-test]-seeder-for-vendor-and-product-vendor

### CODE REVIEW

File: internal/product/product.go [hunk 1]
```diff
@@ -54,8 +53,8 @@ type UOM struct {
 }
 
 type ProductVendor struct {
-	Product
-	vendors.Vendor
+	ProductID string `db:"product_id" json:"product_id"`
+	VendorID  string `db:"vendor_id" json:"vendor_id"`
```
Comment (edutjie) [inline internal/product/product.go:57]: Is there any reason rewriting this as VendorID and ProductID, instead of the object?

**Comment by edutjie** [comment]
Since this is a new development tool and Vale hasn't documented in README how to run the seeder in the previous PR. Could you kindly make the README in this PR to help the developers understand how to run the seeder?

File: scripts/seeder/main.go [hunk 2]
```diff
@@ -22,15 +23,18 @@ const (
 	productCategoryFixtureFile = "./fixtures/product/product_category.jsonc"
 	productTypeFixtureFile     = "./fixtures/product/product_type.jsonc"
 	uomFixtureFile             = "./fixtures/product/uom.jsonc"
+	vendorFixtureFile          = "./fixtures/vendor/vendor.jsonc"
```
Comment (valeelim) [inline scripts/seeder/main.go:26]: i think the file isn't pushed yet?

File: scripts/seeder/main.go [hunk 3]
```diff
@@ -48,8 +52,12 @@ func main() {
 		seedProductType(ctx)
 	case "uom":
 		seedUOM(ctx)
+	case "vendor":
+		seedvendor(ctx)
```
Comment (valeelim) [inline scripts/seeder/main.go:56]:
```suggestion
		seedVendor(ctx)
```

File: scripts/seeder/main.go [hunk 4]
```diff
@@ -164,6 +176,54 @@ func seedUOM(ctx context.Context) {
...
+		log.Println("Error unmarshalling")
+		panic(err)
+	}
+
+	// for _, tempVendor := range temp {
+	// 	theVendor := tempVendor.Vendor
+	// 	theVendor.ModifiedDate, _ = time.Parse(time.DateTime, tempVendor.ModifiedDate)
+	// 	theVendor.Date, _ = time.Parse(time.DateOnly, tempVendor.Date)
+	// 	listOfVendor = append(listOfVendor, theVendor)
+	// }
```
Comment (valeelim) [inline scripts/seeder/main.go:220]: we can remove these

File: scripts/seeder/main.go [hunk 5]
```diff
@@ -164,6 +176,54 @@ func seedUOM(ctx context.Context) {
...
+
+	if err := vendorSeeder.SetupVendors(ctx, listOfVendor); err != nil {
+		panic(err)
+	}
+}
+
+func seedProductVendor(ctx context.Context) {
+	var listOfProductVendor []product.ProductVendor
+
+	// var temp []product.ProductVendor
```
Comment (valeelim) [inline scripts/seeder/main.go:208]: we can remove this

File: scripts/seeder/main.go [hunk 6]
```diff
@@ -48,8 +52,12 @@ func main() {
 		seedProductType(ctx)
 	case "uom":
 		seedUOM(ctx)
+	case "vendor":
+		seedvendor(ctx)
+	case "product-vendor":
+		seedProductVendor(ctx)
 	default:
-		log.Println("Usage: go run scripts/seeder/main.go [product|product_category|product_type|uom].")
+		log.Println("Usage: go run scripts/seeder/main.go [product|product_category|product_type|uom|vendor|product-vendor].")
```
Comment (valeelim) [inline scripts/seeder/main.go:60]: i think we can have a convention either using underscore or dash, I don't really have a preference to any, can you adjust them?

File: internal/product/product.go (hunk 1 above)
Comment (valenciusap17) [reply internal/product/product.go:57]: I don't see any use case that needed both the object. Additionally, I just need to store both the ID and the data type name is relevant, matching the table.  Later, if we need to retrieve both objects after doing sql query, I believe would be better to create a new custom return data type for that.

Comment (edutjie) [reply internal/product/product.go:57]: I see, got it! Thanks

**Review by edutjie — APPROVED** [review]
LGTM

**Comment by FreeJ1nG** [comment]
All the things i wanted to say has been said by the other guys :), good job!

**Comment by FreeJ1nG** [comment]
I tried running `make seed-vendor` and it works, FYI

File: Makefile [hunk 7]
```diff
@@ -27,6 +27,10 @@ seed-product-type:
 	@go run scripts/seeder/main.go product_type
 seed-uom:
 	@go run scripts/seeder/main.go uom
+seed-vendor:
+	@go run scripts/seeder/main.go vendor
+seed-product-vendor:
+	@go run scripts/seeder/main.go product-vendor
```
[the remark this answers is not in the document]
Comment (valenciusap17) [reply Makefile:33]: whoaa my bad ✌️

**Review by valeelim — APPROVED** [review]
LGTM

**Pull request merged**